src/main/java/traincontroller/Train.java

```diff
@@ -0,0 +1,130 @@
+/*
```

Comment:
In the same vein as comments I made on mitch's and issac's PR's It's a bit odd to have your own train class that acts as a stand-in for the real train model. I would recommend using the mockito framework (https://github.com/mockito/mockito) to make "fake" or "mock" objects that you can use during testing instead of some custom train model class.